src/binding/julia/defs.hpp

```diff
@@ -0,0 +1,221 @@
+#ifndef DEFS_HPP
+#define DEFS_HPP
+
+#include "openPMD/openPMD.hpp"
+
+#include <jlcxx/array.hpp>
+#include <jlcxx/jlcxx.hpp>
+#include <jlcxx/module.hpp>
+#include <jlcxx/stl.hpp>
+#include <jlcxx/tuple.hpp>
+#include <jlcxx/type_conversion.hpp>
+
+#include <algorithm>
+#include <array>
+#include <complex>
+#include <map>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+using namespace openPMD;
+
+template <typename T> using array7 = std::array<T, 7>;
+
+// Generate code fo all openPMD types. Use is e.g. as follows:
+//   #define USE_TYPE(NAME, ENUM, TYPE) \
+//     type.method("get_" NAME, &Attribute::get<TYPE>);
+//   FORALL_OPENPMD_TYPES
+//   #undef USE_TYPE
+//
+// We disable `long double` since Julia does not support this type
+// We disable `long double` since Julia does not support this type
+#define FORALL_OPENPMD_TYPES                                                   \
```

Comment:
FYI: We generally use templates for such things, e.g. `switchType` found in `include/openPMD/DatatypeHelpers.hpp`. But otherwise, I think we can leave this as macros for now (especially since `switchType` is not yet as nice to use as `std::visit`, but could be as it does pretty much the same thing).
Otherwise, for macros, you might want to consider passing `USE_TYPE` as a macro argument as the ADIOS2 team [does](https://github.com/ornladios/ADIOS2/blob/master/source/adios2/common/ADIOSMacros.h), otherwise you might get surprising behavior for nested calls.

Reply:
I am now passing `USE_TYPE` as macro argument.